src/js/actions/tools.js

```diff
@@ -727,6 +745,43 @@ define(function (require, exports) {
     enterPathModalState.modal = true;
 
     /**
+     * recreate pointer policies if we're in Vector Mask Mode on UI Store change events
+     *
+     * return {Promise}
+     */
+    var handleVectorMaskUIChange = function () {
+        var toolStore = this.flux.store("tool"),
```

Comment:
You're reading state from the tool store here, so you need at least a `JS_TOOL` read lock.